python/ray/includes/gcs_client.pxi

```diff
@@ -595,6 +745,77 @@ cdef raise_or_return(tup):
 # Must not raise exceptions, or it crashes the process.
 #############################################################
 
+# IDEA(ryw): use Cython fused types to reduce boilerplate.
+cdef convert_optional_raw_get_all_actor_info_reply(
+        CRayStatus status, optional[CGetAllActorInfoReply]&& c_data) with gil:
+    # -> gcs_service_pb2.GetAllActorInfoReply
+    cdef c_string serialized_reply
+    try:
+        check_status_timeout_as_rpc_error(status)
+        with nogil:
+            serialized_reply = c_data.value().SerializeAsString()
+        proto = gcs_service_pb2.GetAllActorInfoReply()
+        proto.ParseFromString(serialized_reply)
```

Comment:
We need to see the overhead of this `SerializeAsString` + `SerializeAsString`.

Challenge: is there a way to avoid doing that?

Reply:
Did a quick look and we don't have an easy way to do this. One can play with grpc for a longer time to make a solution but I don't think it worths the dev time.

Reply:
Can we do some benchmarks to measure the overhead?

Reply:
yeah this can potentially overload dashboard?